packages/frontend/src/components/InputCreator/BooleanRadio.tsx

```diff
@@ -0,0 +1,116 @@
+import { IFieldBooleanRadioOption } from '@plumber/types'
+
+import { Controller, useFormContext } from 'react-hook-form'
+import Markdown from 'react-markdown'
+import { FormControl, RadioGroup, Stack } from '@chakra-ui/react'
+import {
+  FormErrorMessage,
+  FormLabel,
+  Radio,
+} from '@opengovsg/design-system-react'
+
+interface BooleanRadioProps {
+  name: string
+  label?: string
+  description?: string
+  required?: boolean
+  defaultValue?: boolean
+  options?: IFieldBooleanRadioOption[]
+}
+
+function convertBooleanToString(value: boolean | null): string {
+  if (value === null) {
+    return ''
+  }
+  return value ? 'yes' : 'no'
+}
+
+function convertStringToBoolean(value: string): boolean | null {
+  if (value === '') {
+    return null
+  }
+  return value === 'yes'
+}
+
+// show no then yes by default
+const defaultLabelOptions: IFieldBooleanRadioOption[] = [
+  {
+    label: 'No',
+    value: false,
+  },
+  {
+    label: 'Yes',
+    value: true,
+  },
+]
+
+export default function BooleanRadio(props: BooleanRadioProps) {
```

Comment:
As I understand it, you created this component to solve the problem where `SingleSelect` cannot store non-string values.

Does `SingleSelect` work for the case where values are numbers instead of booleans (e.g. scheduler)?

Reply:
damn good catch sir, hmm wah you are right, 0 won't work because it is `falsy`, ok lemme catch u in the office later to discuss about this